packages/@aws-cdk/aws-dynamodb/lib/replica-provider.ts

```diff
@@ -0,0 +1,79 @@
+import { NestedStack } from '@aws-cdk/aws-cloudformation';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Construct, Duration, Stack } from '@aws-cdk/core';
+import * as cr from '@aws-cdk/custom-resources';
+import * as path from 'path';
+
+export class ReplicaProvider extends NestedStack {
+  /**
+   * Creates a stack-singleton resource provider nested stack.
+   */
+  public static getOrCreate(scope: Construct) {
+    const stack = Stack.of(scope);
+    const uid = '@aws-cdk/aws-dynamodb.ReplicaProvider';
+    return stack.node.tryFindChild(uid) as ReplicaProvider || new ReplicaProvider(stack, uid);
+  }
+
+  /**
+   * The custom resource provider.
+   */
+  public readonly provider: cr.Provider;
+
+  /**
+   * The onEvent handler
+   */
+  public readonly onEventHandler: lambda.Function;
+
+  /**
+   * The isComplete handler
+   */
+  public readonly isCompleteHandler: lambda.Function;
```

Comment:
Why do we need these public?

Reply:
allows to use the `grant` api in `table.ts`:

```ts
    // Permissions in the source region
    this.grant(provider.onEventHandler, 'dynamodb:*');
    this.grant(provider.isCompleteHandler, 'dynamodb:DescribeTable');
```

Reply:
OK cool